src/Couchbase.Lite.Shared/API/Sync/Replicator.cs

```diff
@@ -265,10 +266,97 @@ public void Stop()
             });
         }
 
+        /// <summary>
+        /// Gets a list of document IDs that are going to be pushed, but have not been pushed yet
+        /// <item type="bullet">
+        /// <description>API is a snapshot and results may change between the time the call was made and the time</description>
+        /// </item>
+        /// </summary>
+        /// <returns>An immutable set of strings, each of which is a document ID</returns>
+        /// <exception cref="CouchbaseLiteException">Thrown if no push replication</exception>
+        /// <exception cref="CouchbaseException">Thrown if an error condition is returned from LiteCore</exception>
+        [NotNull]
+        public IImmutableSet<string> GetPendingDocumentIDs()
+        {
+            if (!IsPushing()) {
+                WriteLog.To.Database.E(Tag, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs);
+                throw new CouchbaseLiteException(C4ErrorCode.Unsupported, CouchbaseLiteErrorMessage.PullOnlyPendingDocIDs);
+            }
+
+            if (_repl == null) {
+                StartInternal();
+            }
+
+            var result = new HashSet<string>();
+            C4Error err;
+            var ids = Native.c4repl_getPendingDocIDs(_repl, &err);
+
+            if (err.code > 0) {
+                WriteLog.To.Database.E(Tag, err.ToString());
+                throw CouchbaseException.Create(err);
+            }
+
+            if (ids != null) {
+                var flval = Native.FLValue_FromData(ids, FLTrust.Trusted);
+                var flarr = Native.FLValue_AsArray(flval);
+                var cnt = (int)Native.FLArray_Count(flarr);
+
+                for (int i = 0; i < cnt; i++) {
+                    var flv = Native.FLArray_Get(flarr, (uint)i);
+                    result.Add(Native.FLValue_AsString(flv));
```

Comment:
also, this might be a `FLString`(Native.FLValue_AsString(flv)), which might need to convert it to `string`

Reply:
Native.FLValue_AsString(flv) converts array item to string.

Reply:
oh is it, sorry about that.